python/pyspark/sql/tests/test_column.py

```diff
@@ -139,6 +139,33 @@ def test_bitwise_operations(self):
         result = df.select(functions.bitwiseNOT(df.b)).collect()[0].asDict()
         self.assertEqual(~75, result['~b'])
 
+    def test_with_field(self):
```

Comment:
I think doctests are good enough for positive cases. Let's just add negative cases by catching exceptions here.

Reply:
Can you elaborate a little bit on this? Trying to figure out what to add/change.

Reply:
Oh I meant, the test cases you added in doctests look good enough as positive cases. So, we can remove this test you added at `test_column.py`.

Instead, I think we can have some tests that checks exception types and messages here at `test_column.py`.

Reply:
Gotcha alright

Reply:
I updated the test, changing the positive case to match the doc string example, and added TypeError checks for the arguments. Did you want more exception checking beyond that?